aws_lambda_powertools/tracing/tracer.py

```diff
@@ -227,7 +257,10 @@ def put_metadata(self, key: str, value: Any, namespace: Optional[str] = None):
 
         namespace = namespace or self.service
         logger.debug(f"Adding metadata on key '{key}' with '{value}' at namespace '{namespace}'")
-        self.provider.put_metadata(key=key, value=value, namespace=namespace)
+        if self._is_custom_provider():
+            self.provider.set_attribute(key=f"{namespace}.{key}", value=value)
+        else:
+            self.provider.put_metadata(key=key, value=value, namespace=namespace)  # type: ignore
```

Comment:
What if you just use `set_attribute` and make sure it works with both annotations and metadata on the Xray provider?

Reply:
same as above